seed_data/management/commands/alter_data.py

```diff
@@ -175,32 +182,50 @@ def clear_future_runs(user=None, course=None):  # pylint: disable=unused-argumen
     return CourseRun.objects.filter(course=course, edx_course_key__startswith=NEW_COURSE_RUN_PREFIX).delete()
 
 
-@accepts_or_calculates_now
-def add_past_failed_run(user=None, course=None, now=None, grade=DEFAULT_FAILED_GRADE, add_if_exists=False):
-    """Adds a past failed course run for a given user and course"""
+def get_past_course_run(user=None, course=None, now=None, add_if_exists=False):
+    """Loop through past course runs and find one without CachedCurrentGrade data"""
     chosen_past_run = None
     past_runs = CourseRun.objects.filter(
         course=course,
         end_date__lt=now,
     ).exclude(end_date=None).order_by('-end_date').all()
-    # Loop through past course runs and find one without CachedCurrentGrade data
     for past_run in past_runs:
-        if not CachedCurrentGradeHandler(user).exists(past_run):
+        if not (CachedCurrentGradeHandler(user).exists(past_run)
+                or CachedCertificateHandler(user).exists(past_run)):
```

Comment:
in the [code sample i posted](https://github.com/mitodl/micromasters/pull/2119#discussion_r92504826), i fixed a minor bug in this loop. this is incorrectly using `continue` at line 196. the effect is that instead of returning the most recent un-graded course run, it returns the oldest. not a big deal, but it can result in unexpected behavior. since this function doesn't need to do anything with the `chosen_past_run` anymore, we can just return it immediately as soon as we find it